### ISSUE TITLE
Output modules that only sometimes work

### ISSUE THREAD

**Issue by timcdlucas** [zoonproject/zoon#171]
It seems inevitable that some output modules will only work sometimes.

`PerformanceMeasures` only works if there is a validation set.

I've been working on a module that implements all the models in [caret](http://topepo.github.io/caret/training.html). One of the great things in `caret` is plots of model performance with changing hyperparameters. These will only ever work on models where hyperparameters have been selected. And it's hard to see how they could generalise over different models that do select hyperparameters (gbm and caret for example). But it's a shame to let that get in the way.

![hyps](https://cloud.githubusercontent.com/assets/3583796/9946284/33ccc624-5d8a-11e5-82f6-4b7a9ba8ece3.png)

How do we want to deal with this in general? It could be that some output modules would apply to some modules used in a list and not others. So an error seems too strong. We might have to require these modules to check if the right type of stuff is coming in and print a warning otherwise (while avoiding errors). I guess this seems best to me at the moment. Though it's another requirement for module writers.

**Comment by goldingn** [comment]
I think `PerformanceMeasures` always works now - it issues a warning that training-set validation is a stupid thing to do.

I agree it's a shame to lose the functionality available in existing R packages. A nicer approach (than an error or warning) might be to define a module which does nice things like this for the caret models, and has some less exciting output for non-caret models. Provided the docs say that's what it'll do, and it doesn't error out, I think that works for all.

**Comment by timcdlucas** [comment]
Think that sounds like a good plan. Trying to think of the variation of inputs coming in. I'll close this. And hope everyone who wanted to weigh in saw it.
